### ISSUE TITLE
The position of the parameter affects the use？ I have three parameters @Header,@QueryMap,@Url. I put "@URL" parameter on the last one useless, put the first one useful.why?

### ISSUE THREAD

**Issue by mtf7101520** [square/retrofit#2470]
What kind of issue is this?

 - [ ] Question. This issue tracker is not the place for questions. If you want to ask how to do
       something, or to understand why something isn't working the way you expect it to, use Stack
       Overflow. https://stackoverflow.com/questions/tagged/retrofit

 - [ ] Bug report. If you’ve found a bug, spend the time to write a failing test. Bugs with tests
       get fixed. Here’s an example: https://gist.github.com/swankjesse/6608b4713ad80988cdc9

 - [ ] Feature Request. Start by telling us what problem you’re trying to solve. Often a solution
       already exists! Don’t send pull requests to implement new features without first getting our
       support. Sometimes we leave features out on purpose to keep the project small.

**Comment by mtf7101520** [comment]
post request had no problem，The get request will go wrong   “ java.lang.NullPointerException: Attempt to invoke virtual method 'okhttp3.HttpUrl$Builder okhttp3.HttpUrl$Builder.addQueryParameter(java.lang.String, java.lang.String)' on a null object reference”

**Comment by JakeWharton** [comment]
Please provide a failing test case or executable sample that reproduces the
problem.

**Comment by mtf7101520** [comment]
It is initialize.
 private ArchiveHttpServiceProxy(String url) {
        HttpLoggingInterceptor logging = new HttpLoggingInterceptor();
        logging.setLevel(HttpLoggingInterceptor.Level.BODY);
        OkHttpClient.Builder builder = new OkHttpClient.Builder()
                .connectTimeout(Config.CONNECT_TIMEOUT, TimeUnit.SECONDS)
                .readTimeout(Config.READ_TIMEOUT, TimeUnit.SECONDS);
        if (BuildConfig.DEBUG) {
            builder.addInterceptor(logging);
        }
        OkHttpClient okHttp = builder.build();
        mService = new Retrofit.Builder()
                .baseUrl(url)
                .addConverterFactory(GsonConverterFactory.create())
                .addCallAdapterFactory(RxJava2CallAdapterFactory.create()) 
                .client(okHttp)
                .build()
                .create(ArchiveHttpService.class);
    }

This is a get request.

 @GET()
    Observable<Result<FileList>> getFiles(@Url String url, @Header("wps-sid") String sid, @QueryMap Map<String, Object> params);

This is where the call is made.

public Observable<Result<FileList>> getFiles(Map<String, Object> params) {
        String sid = getWpsSid();
        if (TextUtils.isEmpty(sid)) {
            LogUtils.e(TAG, "getFiles wps-sid is empty");
            return null;
        }
        try {
            return mService.getFiles(URLMapController.getDTFilesUrl(), sid, params);
        } catch (Exception e) {
            LogUtils.e(TAG, "getFiles error: " + e.getMessage());
            return null;
        }
    }

That's right，But when i write it like that。

 @GET()
 Observable<Result<FileList>> getFiles(@Header("wps-sid") String sid, @QueryMap Map<String, Object> params，@Url String url);

it will be given an error java.lang.NullPointerException: Attempt to invoke virtual method 'okhttp3.HttpUrl$Builder okhttp3.HttpUrl$Builder.addQueryParameter(java.lang.String, java.lang.String)' on a null object reference
But the post request is no problem。This bug bothers me for a long time,

**Comment by JakeWharton** [comment]
Yes @Url needs to be first, but we have a check that ensure this is the
case and a test that ensures the check works. I'll have to try it soon and
see what's going on.
